File: L6_Odd_Even_Linked_List.java
class Node{
    public int data;
    public Node next;

    Node()
    {
        this.data = 0;
        this.next = null;
    }

    Node(int data)
    {
        this.data = data;
        this.next = null;
    }
}


public class L6_Odd_Even_Linked_List {
    public static Node oddEvenList(Node head) {
        // Write your code here.
        if(head.next == null) return head;
        Node odd = head;
        Node even = head.next;
        Node evenHead = even;
        while(even.next != null){
            odd.next = even.next;
            even.next = even.next.next;
            if(odd.next != null) odd = odd.next;
            if(even.next != null) even = even.next;
        }

        odd.next = evenHead;
        if(even != null) even.next = null;
        return head;
    }
}

File: L6_Odd_Even_Linked_List.cpp
//Following is the Linked list node structure already written

class Node{
public:
    int data;
    Node* next;
    Node(int data)
    { 
        this->data = data;
        this->next = nullptr;
    }
};

Node* oddEvenList(Node* head){
	// Write your code here.
    if(!head->next) return head;
    Node *odd = head;
    Node *even = head->next;
    Node *evenHead = even;

    while(even->next){
        odd->next = even->next;
        even->next = even->next->next;
        if(odd->next) odd = odd->next;
        if(even->next) even = even->next;
    }

    odd->next = evenHead;
    if(even) even->next = nullptr;
    return head;
}